common/static/common/js/discussion/discussion_router.js

```diff
@@ -129,9 +128,7 @@
             DiscussionRouter.prototype.navigateToThread = function(thread_id) {
                 var thread;
                 thread = this.discussion.get(thread_id);
-                return this.navigate("" + (thread.get("commentable_id")) + "/threads/" + thread_id, {
-                    trigger: true
-                });
+                return this.showThread(thread);
```

Comment:
@bradenmacdonald This had to be changed because for some reason, `showThread` does not have access to an up-to-date thread collection, so it couldn't show a newly added thread.

(To be more specific, `showThread` would try to retrieve the thread via `this.discussion.get(thread_id)`, using `thread_id` from the `:forum_name/threads/:thread_id` URL. However, `this.discussion` did not contain an entry for the newly added thread, so the lookup would fail).